src/mcp23017.cpp

```diff
@@ -36,6 +99,20 @@ int Mcp23017::begin(uint8_t directions[8]) {
     int rc;
 
     // TODO: Add device ID check
+    Wire.requestFrom(this->addr, 1);
+    if (Wire.available()){
+        rc = Wire.read();
+    }
+    else{
+        return 0;
+    }
+
+    int c = 0;
+
+    while (c < 8){
```

Comment:
It would probably be better to use a for loop here for conciseness (1 line for the initialization, boolean check, and incrementing).